go/vt/discovery/healthcheck.go

```diff
@@ -448,8 +453,7 @@ func (hc *HealthCheckImpl) updateHealth(th *TabletHealth, prevTarget *query.Targ
 			}
 		}
 	case isPrimary && !isPrimaryUp:
-		// No healthy master tablet
-		hc.healthy[targetKey] = []*TabletHealth{}
+		hc.deleteHealthDateLocked(targetKey, tabletAlias)
```

Comment:
I don't believe this is correct. We want to delete it from `healthy` but not from `healthData`.

Reply:
Updated to keep it same as the original logic.